File: functional-programming-in-scala/Functional-progamming-in-scala/Chap_2/partial.scala
//
// partial1 is a function that accepts two parameters: -
//  - a of type A
//  - function that accepts types A,B and returns C
//
//  The function returns a function that accepts B and returns C.
//
//
def partial1[A,B,C](a: A, f: (A,B) => C): B => C = 
  b => f(a,b)


val add1 = partial1(1, (a: Int, b: Int) => a + b)

println(add1(10))

val prefixHello = partial1("Hello ",(a: String, b: String) => a + b)

println(prefixHello("Martin"))

File: functional-programming-in-scala/Functional-progamming-in-scala/Chap_2/partial.py
#
# Python version of partial function
#
from __future__ import print_function

def partial1(a,func):
    return lambda b: func(a,b)

add1 = partial1(1,lambda  a,b: a+b)

print(add1(1))

prefixHello = partial1("Hello ",lambda a,b: a+b)

print(prefixHello("Martin"))
